arrow/flight/flightsql/example/sql_batch_reader.go

```diff
@@ -264,7 +265,7 @@ func (r *SqlBatchReader) Next() bool {
 
 	rows := 0
 	for rows < maxBatchSize && r.rows.Next() {
-		if err := r.rows.Scan(r.rowdest...); err != nil {
+		if err := r.rows.Scan(r.rowdest...); err != nil && err.Error() != io.EOF.Error() {
```

Comment:
Huh...it's quite weird that this can return EOF

Reply:
From what I can tell, it's something weird with the sqlite handler implementation. it shouldn't be returning EOF, which looks like it isn't even returning `io.EOF` and instead just returns the string "EOF" it's really weird.